core/src/main/scala/org/apache/gearpump/util/Graph.scala

```diff
@@ -165,7 +181,7 @@ class Graph[N, E](vertexList: List[N], edgeList: List[(N, E, N)]) extends Serial
    * edges connected to node
    */
   def edgesOf(node: N): List[(N, E, N)] = {
-    (incomingEdgesOf(node) ++ outgoingEdgesOf(node)).toSet[(N, E, N)].toList.sortBy(_indexs(_))
+    (incomingEdgesOf(node) ++ outgoingEdgesOf(node)).distinct.sortBy(_indexs(_))
```

Comment:
why do we need to `distinct` and `sort` here ?

Reply:
```
 // This is used to ensure the output of this Graph is always stable
 // Like method vertices(), or edges()
 private var _indexs = Map.empty[Any, Int]
````